test/integration/upstream_http_filter_integration_test.cc

```diff
@@ -158,6 +158,25 @@ TEST_P(StaticRouterOrClusterFiltersIntegrationTest, BasicSuccess) {
   expectHeaderKeyAndValue(headers, default_header_key_, default_header_value_);
 }
 
+TEST_P(StaticRouterOrClusterFiltersIntegrationTest, BasicSuccessOnAsyncClient) {
+  GTEST_SKIP() << "TODO(wbpcode): it is expected that AsyncClient requests pass through both "
+                  "upstream_filters chains, but this test currently passes if the filter is in the "
+                  "cluster chain, and fails if it's in the router chain.";
+  // This filter intercepts the downstream request and reissues it via AsyncClient,
```

Comment:
This is expected behavior because the router filter's configuration in the main http filter chain shouldn't effect the side stream.